### PR TITLE
Optimize dot plot rendering

### PR DESCRIPTION
**Pull request by cmdcolin**
This thread proposes an initial speedup for dotplot rendering

Currently there is a large performance drop with accessing observables when useStaticRendering is on

See https://github.com/mobxjs/mobx-react/issues/526

The use of observables is hit hard by cache misses on the server side rendering

This PR proposes to not use the observable, though we could also consider some of the recommendations from that thread

Current main branch, time to render Grape vs Peach PAF - 2.8s
This branch, time to render Grape vs Peach PAF - 100ms

25x speedup with this PR

### CODE REVIEW

**Comment by cmdcolin** [comment]
Also tries a scheme to color forward and reverse strand alignments, which is how many dotplot programs like mummerplot do things

File: plugins/dotplot-view/src/DotplotRenderer/configSchema.ts [hunk 1]
```diff
@@ -16,10 +16,20 @@ export default ConfigurationSchema(
...
+    posColor: {
+      type: 'color',
+      description: 'the color for forward alignments',
+      defaultValue: 'purple',
+    },
+    negColor: {
+      type: 'color',
+      description: 'the color for reverse alignments',
+      defaultValue: 'blue',
+    },
```
Comment (rbuels) [inline plugins/dotplot-view/src/DotplotRenderer/configSchema.ts:28]: do these descriptions need to be updated? they talk about alignments

Comment (cmdcolin) [reply plugins/dotplot-view/src/DotplotRenderer/configSchema.ts:28]: dotplot plots alignments :)

**Comment by cmdcolin** [comment]
I found that even when I add the workaround (putting the code inside a reaction) the process of "unboxing" from mobx is quite a bit less fast than the approach from this PR (operating on snapshots / plain js objects)

**Comment by cmdcolin** [comment]
this PR no longer does the different stranded coloring, as I wanted to get the optimization isolated

**Pull request merged**